src/main/java/org/broadinstitute/hellbender/tools/copynumber/utils/CombineSegmentBreakpoints.java

```diff
@@ -78,8 +80,11 @@
     @Override
     public void traverse() {
 
-        final List<SimpleAnnotatedGenomicRegion> segments1 = SimpleAnnotatedGenomicRegion.readAnnotatedRegions(segmentFiles.get(0), columnsOfInterest);
-        final List<SimpleAnnotatedGenomicRegion> segments2 = SimpleAnnotatedGenomicRegion.readAnnotatedRegions(segmentFiles.get(1), columnsOfInterest);
+        final SimpleAnnotatedGenomicRegionCollection simpleAnnotatedGenomicRegionCollection1 = SimpleAnnotatedGenomicRegionCollection.readAnnotatedRegions(segmentFiles.get(0), columnsOfInterest);
```

Comment:
I think I made the same comment in a previous PR, but I think you will be able to use the collections framework much easier if you just specify the columns you want from each file separately, i.e., `columns-of-interest-for-first-input` and `columns-of-interest-for-second-input` or something along those lines.

You then create an AnnotatedIntervalCollection (after modifying that class appropriately) for each column set, dynamically generating codecs, and read in each corresponding file.  Incorrectly specified or missing columns will be handled by TableReader.

You then create an AnnotatedIntervalCollection for the combination of the two column sets.  You will have to take care to add suffixes when columns are duplicated.  You could extract this logic to some method that combines two AnnotatedIntervalCollections.

Your CLI should then be much simpler---just read in the files, create a new collection by calling the combine method, and then write the new collection.

Reply:
I will look into Annotated Intervals and get back to you.

Reply:
To be clear, you will have to make some modifications to AnnotatedIntervalCollection and its associated classes.

If you do things right, we could probably retool most of the other CNV collections to be AnnotatedIntervalCollections, since that is essentially what they are.  We can do this later, though.